src/Core/Form/IdentifiableObject/DataHandler/CarrierFormDataHandler.php

```diff
@@ -57,6 +57,11 @@ public function create(array $data)
             $data['general_settings']['tracking_url'] ?? '',
             0, // @todo: should not be in the add command but auto-computed or at least be optional
             (bool) $data['general_settings']['active'],
+            (bool) $data['shipping_settings']['has_additional_handling_fee'],
```

Comment:
The data is ok, but the form will fail once this is merged since the form type has no such data. I guess it's acceptable for now but we must keep that in mind